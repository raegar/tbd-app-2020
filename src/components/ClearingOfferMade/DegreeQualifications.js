import React, { useState } from 'react';
import "./DegreeQualifications.css"; 
import {Row, Col, Container} from "react-bootstrap";
import RadioButton from '../RadioButton/RadioButton';
import TextBox from '../TextBox/TextBox';
import UCPButton from '../Buttons/UCPButton';

function DegreeQualifications() {
	const [levelFour, setLevelFour] = useState(false);
	const [nameGrade, setNameGrade] = useState("");
	const [recentSchool, setRecentSchool] = useState("");
	const [priorYesState, setPriorYesState] = useState({
		checked: false,
		value: "",
	});
	const [priorNoState, setPriorNoState] = useState({
		checked: false,
		value: "",
	});
	const [elqYesState, setElqYesState] = useState({
		checked: false,
		value: "",
	});
	const [elqNoState, setElqNoState] = useState({
		checked: false,
		value: "",
	});

	function handleClick(e) {
    	setLevelFour(e === 'yes');
	}

	function onUpdateNameGrade(e) {
		setNameGrade(e.target.value);
	}

	function onUpdateRecentSchool(e) {
		setRecentSchool(e.target.value);
	}

	function onPriorYesChanged(e) {
		setPriorYesState({
			checked: e.target.checked,
			value: e.target.value,
		});
	}

	function onPriorNoChanged(e) {
		setPriorNoState({
			checked: e.target.checked,
			value: e.target.value,
		});
	}

	function elqYesChanged(e) {
		setElqYesState({
			checked: e.target.checked,
			value: e.target.value,
		});
	}

	function elqNoChanged(e) {
		setElqNoState({
			checked: e.target.checked,
			value: e.target.value,
		});
	}

	const PriorLevelFour = [
		{label: 'Yes', id: 'prior-yes', value: 'yes', handleChange: onPriorYesChanged},
		{label: 'No', id: 'prior-no', value: 'no', handleChange: onPriorNoChanged}
	];
	
	const ELQ = [
		{label: 'Yes', id: 'elq-yes', value: 'yes', handleChange: elqYesChanged},
		{label: 'No', id: 'elq-no', value: 'no', handleChange: elqNoChanged}
	];

	function saveSelectedData() {
    	if (priorYesState.checked) {
    		global.ApplicationFormData.anyLevelFourQualification = priorYesState.value;
    		global.ApplicationFormData.levelFourQualificationDetails = nameGrade;
    	}

    	if (priorNoState.checked) {
    		global.ApplicationFormData.anyLevelFourQualification = priorNoState.value;
    	}  

    	global.ApplicationFormData.mostRecentSchool = recentSchool;

    	if (elqYesState.checked) {
    		global.ApplicationFormData.elqQualification = elqYesState.value;
    	}

    	if (elqNoState.checked) {
    		global.ApplicationFormData.elqQualification = elqNoState.value;
    	}

    	console.log(global.ApplicationFormData);
	}

	let nextpage;

	if (global.userType === "staff") {
		nextpage = "/ClearingOfferMade";
	} else {
		nextpage = "/";
	}

	return (
		<>
			<Container className="degree-form d-flex justify-content-center">
				<link 
					href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500;700&display=swap" 
					rel="stylesheet">
				</link>
				<div className="form-background">
					<h3 className="form-title">Level 4 or Above Experience</h3>
					<Row>
						<Col>
							<RadioButton
								subtitle="Do you have any level 4 or above qualifications?"
								name="level-four"
								options={PriorLevelFour}
								oc={handleClick}
								data-testid="level-4-radio"
							/>
							{
								(levelFour)
									? <TextBox 
										id="nameGrade" 
										title="Enter qualification name and grade" 
										name="l-4-qual" 
										size={30}
										onChange={onUpdateNameGrade}
									/>
									: <span></span>
							}

						</Col>
						<Col>
							<TextBox 
								id="schoolDetail"
								name="recent-school" 
								title="What is the name of the most recent school/college attended?" 
								size={25} 
								type="text"
								onChange={onUpdateRecentSchool}
							/>
							<RadioButton 
								subtitle={"ELQ - is the applicant applying to study a qualification that is equivalent or" +
								" lower to one they already hold?"}
								options={ELQ}
								name="elq"
							/>
						</Col>
					</Row>
				</div>
			</Container>
			<Container>
				<Row id="buttonRow">
					<Col className="centered-buttons">
						<UCPButton to='/LevelThreeQualifications'
							primary="True"
							className="mediumbutton"
							buttonText="Go Back"
						/>
					</Col>
					<Col className="centered-buttons" onClick={saveSelectedData}>
						<div id="confirmButton">
							<UCPButton
								to={nextpage}
								primary="True"
								className="mediumbutton"
								buttonText="Confirm and Go"
							/>
						</div>
					</Col>
				</Row>
			</Container>
		</>
	);
}

export default DegreeQualifications;