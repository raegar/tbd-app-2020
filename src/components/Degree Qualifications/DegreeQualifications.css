.degree-form {
    padding-top: 20px;
    font-family: 'Montserrat', sans-serif;
}

.form-background {
    width: 80%;
    background-color: #B2DDDF;
    display: inline-block;
    padding: 20px;
    /*text-align: left;*/
    border: 1px solid black;
    margin-bottom: 10px;
}
.centered-buttons {
    text-align: center;
}

.form-container {
    padding-right: 0px !important;
    padding-left: 0px !important;
}

.form-title {
    text-align: center;
}
.form-right {
    float: right;
    width: 50%;
    margin-top: 30px;
}
.form-left {
    float:left;
    width: 50%;
    padding-left: 40px;
}

.mobiletextbox input{
    width: 250px;
    text-align: center;
}
.mobiletextbox {
    width: 200px;
}
.mobiletextbox label {
    margin-left: 5px;
}

@media only screen and (max-width: 768px){
    .form-left {
    width: 80%;
    float: left;
    padding-left: 0px;
    }

    .form-right {
        width: 100%;
        float: left;
    }
}

.h3 {
    font-family: 'Montserrat', sans-serif;
}