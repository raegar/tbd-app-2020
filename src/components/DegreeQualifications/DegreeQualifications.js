import React from 'react';
import "./DegreeQualifications.css"; 
import {Row, Col, Container} from "react-bootstrap";
import RadioButton from '../RadioButton/RadioButton';
import TextBox from '../TextBox/TextBox';
import UCPButton from '../Buttons/UCPButton';

const ELQ = [
	{label:'Yes', id:'elq-yes', value:"yes"},
	{label:'No', id:'elq-no', value:"no"}
];

const PriorLevelFour = [
	{label: 'Yes', id: 'prior-yes', value: 'yes'},
	{label: 'No', id: 'prior-no', value: 'no'}
];

class DegreeQualifications extends React.Component {
	constructor(props) {
		super(props);
		this.props = props;

		this.state = {
			levelFour: false
		};
	}

    setLevelFour = (value) => {
    	this.setState({levelFour: value});
    }

    handleClick = (e) => {
    	this.setLevelFour(e === 'yes');
    }

    saveSelectedData = () => {
    	//TODO: make this Reacty because this is bad.
    	if (document.getElementById("prior-yes").checked) {
    		global.ApplicationFormData.anyLevelFourQualification = document.getElementById("prior-yes").value;
    		global.ApplicationFormData.levelFourQualificationDetails = document.getElementById("nameGrade").value;
    	}

    	if (document.getElementById("prior-no").checked) {
    		global.ApplicationFormData.AnyLevelFourQualification = document.getElementById("prior-no").value;
    	}

    	global.ApplicationFormData.mostRecentSchool = document.getElementById("schoolDetail").value;

    	if (document.getElementById("elq-yes").checked) {
    		global.ApplicationFormData.elqQualification = document.getElementById("elq-yes").value;
    	}

    	if (document.getElementById("elq-no").checked) {
    		global.ApplicationFormData.elqQualification = document.getElementById("elq-no").value;
    	}

    	console.log(global.ApplicationFormData);
    }

    render = () => {
    	const radioButtonSubtitle = 
            "ELQ - is the applicant applying to study a qualification that is equivalent or lower to one they already hold?";
    	let nextpage;

    	if (global.userType === "staff") {
    		nextpage = "/ClearingOfferMade" ;
    	} else {
    		nextpage = "/ClearingOfferMadeStudentEnd";
    	}

    	return (
    		<>
    			<Container className="degree-form d-flex justify-content-center">
    				<link 
    					href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500;700&display=swap" 
    					rel="stylesheet"
    				/>
    				<div className="form-background">
    					<h3 className="form-title">Level 4 or Above Experience</h3>
    					<Row>
    						<Col>
    							<RadioButton
    								subtitle="Do you have any level 4 or above qualifications?"
    								name="level-four"
    								options={PriorLevelFour}
    								oc={this.handleClick}
    								data-testid="level-4-radio"
    							/>
    							{
    								(this.state.levelFour)
    									? <TextBox 
    										id="nameGrade" 
    										title="Enter qualification name and grade" 
    										name="l-4-qual" 
    										size={30} 
    									/>
    									: <span></span>
    							}
    
    						</Col>
    						<Col>
    							<TextBox 
    								id="schoolDetail"
    								name="recent-school" 
    								title="What is the name of the most recent school/college attended?" 
    								size={25} 
    								type="text" 
    							/>
    							<RadioButton 
    								subtitle={radioButtonSubtitle}
    								options={ELQ}
    								name="elq"
    							/>
    						</Col>
    					</Row>
    				</div>
    			</Container>
    			<Container>
    				<Row id="buttonRow">
    					<Col className="centered-buttons">
    						<UCPButton to='/LevelThreeQualifications'
    							primary="True"
    							className="mediumbutton"
    							buttonText="Go Back"
    						/>
    					</Col>
    					<Col className="centered-buttons" onClick={this.saveSelectedData}>
    						<div id="confirmButton">
    							<UCPButton
    								to={nextpage}
    								primary="True"
    								className="mediumbutton"
    								buttonText="Confirm and Go"
    							/>
    						</div>
    					</Col>
    				</Row>
    			</Container>
    		</>
    	);
    }
}

export default DegreeQualifications;