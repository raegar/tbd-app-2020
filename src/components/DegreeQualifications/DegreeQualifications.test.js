import React from 'react';
import { render, cleanup, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import DegreeQualifications from './DegreeQualifications';
import { BrowserRouter as Router} from 'react-router-dom';

//cleans up 
afterEach(cleanup);

it("Renders Degree Qual", () => {
    const { asFragment } = render(<Router><DegreeQualifications /></Router>);
    expect(asFragment()).toMatchSnapshot();
});

test("Level 4 Qualification Conditional rendering", () => {
    render(
        <Router>
            <DegreeQualifications />
        </Router>
    )

    expect(screen.getByTestId('prior-yes')).toBeInTheDocument()
    expect(screen.queryByText('Enter qualification name and grade')).toBe(null)

    fireEvent.click(screen.getByTestId('prior-yes'))

    expect(screen.queryByText('Enter qualification name and grade')).toBeInTheDocument()

    fireEvent.click(screen.getByTestId('prior-no'))

    expect(screen.queryByText('Enter qualification name and grade')).toBe(null)
})