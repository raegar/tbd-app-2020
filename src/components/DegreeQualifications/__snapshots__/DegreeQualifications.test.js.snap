// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Degree Qual 1`] = `
<DocumentFragment>
  <div>
    <div>
      <header
        class="Header"
      >
        <a
          href="/tbd-app-2020"
        >
          <img
            alt="UCP official logo"
            class="UCPLogoBlue"
            data-testid="ucpLogoHeader"
            src="Logo_White.png"
          />
        </a>
      </header>
      <div
        class="Line"
      />
    </div>
    <div
      style="height: 30px; width: 90%; background-color: rgb(178, 221, 223); border-radius: 50px; margin: 50px;"
    >
      <div
        style="height: 100%; width: 90%; background-color: rgb(0, 92, 110); border-radius: inherit; text-align: right;"
      >
        <span
          style="padding: 5px; color: rgb(255, 255, 255); font-weight: bold;"
        >
           90% 
        </span>
      </div>
    </div>
    <div
      class="degree-form d-flex justify-content-center container"
    >
      <link
        href="https://fonts.googleapis.com/css2?family=Montserrat:wght@500;700&display=swap"
        rel="stylesheet"
      />
      <div
        class="form-background"
      >
        <h3
          class="form-title"
        >
          Level 4 or Above Experience
        </h3>
        <div
          class="row"
        >
          <div
            class="col"
          >
            <div
              class="radio-buttons"
            >
              <h4 />
              <p>
                Do you have any level 4 or above qualifications?
              </p>
              <div
                class="radio-group"
              >
                <input
                  data-testid="prior-yes"
                  id="prior-yes"
                  name="level-four"
                  type="radio"
                  value="yes"
                />
                <label
                  for="prior-yes"
                >
                  Yes
                </label>
              </div>
              <div
                class="radio-group"
              >
                <input
                  data-testid="prior-no"
                  id="prior-no"
                  name="level-four"
                  type="radio"
                  value="no"
                />
                <label
                  for="prior-no"
                >
                  No
                </label>
              </div>
            </div>
            <span />
          </div>
          <div
            class="col"
          >
            <div
              class="radio-buttons"
            >
              <h4 />
              <p>
                ELQ - is the applicant applying to study a qualification that is equivalent or lower to one they already hold?
              </p>
              <div
                class="radio-group"
              >
                <input
                  data-testid="elq-yes"
                  id="elq-yes"
                  name="elq"
                  type="radio"
                  value="yes"
                />
                <label
                  for="elq-yes"
                >
                  Yes
                </label>
              </div>
              <div
                class="radio-group"
              >
                <input
                  data-testid="elq-no"
                  id="elq-no"
                  name="elq"
                  type="radio"
                  value="no"
                />
                <label
                  for="elq-no"
                >
                  No
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="container"
    >
      <div
        class="row"
        id="buttonRow"
      >
        <div
          class="centered-buttons col"
        >
          <a
            href="/LevelThreeQualifications"
          >
            <button
              class="mediumbutton"
              data-testid="ucpButton"
              primary="True"
              type="button"
            >
              Go Back​
            </button>
          </a>
        </div>
        <div
          class="centered-buttons col"
        >
          <div
            id="confirmButton"
          >
            <a
              href="/ClearingOfferMadeStudentEnd"
            >
              <button
                class="mediumbutton"
                data-testid="ucpButton"
                primary="True"
                type="button"
              >
                Confirm and Go​
              </button>
            </a>
          </div>
        </div>
      </div>
    </div>
    <footer
      class="ucpFooter"
    />
  </div>
</DocumentFragment>
`;
